[jjiayyingtt] iP

jjiayyingtt opened the pull request:
# Duke Project

> DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here] [(https://nus-cs2103-Ay2223s2.github.io/website/schedule/week4/project.html)].
2. double-click it.
3.add your tasks.
4. let it manage your tasks for you 😉
And it is FREE!

Features:

 - [X] Managing tasks
 - [X] Managing deadlines
 - [ ] Reminders (coming soon)

Commit:
Add Gradle support

jjiayyingtt committed:
Greet, Echo, Exit

jjiayyingtt committed:
Add, List

jjiayyingtt committed:
Mark as Done

jjiayyingtt committed:
ToDos, Events, Deadlines

jjiayyingtt committed:
Automated Text UI Testing

jjiayyingtt committed:
Handle Errors

jjiayyingtt committed:
Delete

jjiayyingtt committed:
Save

c0j0s commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,180 @@
+                operationHandler("deadline" + dataArr[2] + "/by" + dataArr[3]);
+            } else if (dataArr[0].equals("event ")) {
+                operationHandler("event" + dataArr[2] + "/from" + dataArr[3] + "/to" + dataArr[4]);
+            }*/
```
Maybe you might want to remove commented codes before committing to make the PR neater.

c0j0s commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,180 @@
+    }
+
+    private static String[] stringProcessor(boolean isDeadline, String s){ // isDeadline = false meaning isEvent
+        if (isDeadline){
```
I have noticed several areas in your code without a space after if/for/method declarations. You might want to be careful as it will be painful when you need to resolve check style issues later on.

c0j0s commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+    public void printTask(){
+        System.out.print("[D]");
+        super.printTask();
+        System.out.println("(by: " + date + ")");
```
Nice, OOP from the start. But you might want to extract IO dependencies from your model classes, it will be easier when you decide to change your ui layer in the future.

c0j0s commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,180 @@
+    public DukeException(String errorMessage){
+        super(errorMessage);
+    }
+}
```
Should this be in its own file?

c0j0s reviewed (commented):
overall looking good, but you might want to standardise your code style to comply with the standards.

jjiayyingtt committed:
Dates and Times

pkpaing commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,198 @@
+                err.printStackTrace();
+            }
+        }
+    }
```
This code is well indented and very readable!

pkpaing commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,198 @@
+
+    private static List<Task> toDoList = new ArrayList<>();
+
+    private static void startUp() {
+        String name = "todo bot";
+        System.out.println("Hello from " + name);
+        System.out.println("------------------------------------");
+        System.out.println("I can help you take care of your daily todos :)");
+        System.out.println("There are 3 types of tasks I can handle.");
+        System.out.println("------------------------------------");
+        System.out.println("1. normal todo, format: ");
+        System.out.println("   todo task");
+        System.out.println("2. deadline, format: ");
+        System.out.println("   deadline task /by yyyy-mm-dd");
+        System.out.println("3. event, format: ");
+        System.out.println("   event task /from yyyy-mm-dd /to yyyy-mm-dd");
+        System.out.println("------------------------------------");
+
+    }
```
Very well formatted print statements, easy to read!

pkpaing reviewed (commented):
Looks good to me, keep it up!

jjiayyingtt committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jjiayyingtt committed:
gradle and OOP

jjiayyingtt committed:
packages, JUnit

jjiayyingtt committed:
JAR

jjiayyingtt committed:
JavaDoc

jjiayyingtt committed:
Coding Standard and Find

jjiayyingtt committed:
GUI, Varargs

jjiayyingtt committed:
Add assertion to Parser for error handling.

Assertions are added to operationHandler, when tasks are added to
the list.

jjiayyingtt committed:
Load and Save list from chatbot

Previous commit version's did not get tasks that were saved and
it did not save tasks that were obtained from chatbot.

Link the data file to the chatbot

jjiayyingtt committed:
A-CodeQuality: Improved code quality for Parser

jjiayyingtt committed:
Snoozing/postponing tasks

Provide a way to easily snooze/postpone/reschedule tasks.

jjiayyingtt committed:
moved UI methods from Duke to UI class

jjiayyingtt committed:
no message

jjiayyingtt committed:
no message

jjiayyingtt committed:
fixed merged issue

jjiayyingtt committed:
improved code quality

jjiayyingtt committed:
improved code quality
moved function within operationHandler out to addTask and Snooze

jjiayyingtt committed:
removed methods that were for text ui
improved code quality

jjiayyingtt committed:
Added screenshot and updated User Guide
